### PR TITLE
Feature/Form Select

### PR DESCRIPTION
**Pull request by kamalthapa1997**
## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by toriroe** [comment]
I believe the name for the commit should be descriptive of the feature/component that's being worked on. So for example, the commit message for this might be "feat: form select component" or something similar. Also, I believe there should be a storybook entry for each component, in a file called "form-select.stories.tsx" (just as an example) that's in the same folder as the other component files. But anyone else please feel free to correct me if I'm wrong

**Review by cgenardini — COMMENTED** [review]
Component looks great but will probably need to be refactored to make it more flexible. I just refactored my form component using Dillons as an example and found it very useful! 
I would pay attention to how he used the ...register and validation.

https://github.com/jahorwitz/drops/pull/5/files#r1552670121

File: src/components/form.Select/form.Select.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
```
Comment (cgenardini) [inline src/components/form.Select/form.Select.stories.tsx:1]: The form story should be in the main form.stories file with the other form components/

File: src/components/form.Select/form.Select.tsx [hunk 2]
```diff
@@ -33,36 +34,46 @@ const FormSelect = ({
...
-        className="m-0 box-border  border shadow-sm rounded-lg grow-0 h-14 p-2  w-80  font-text"
-      >
-        <option className="opacity-50	" value="" disabled hidden>
-          {placeholderValue}
-        </option>
-
-        {options.map((option) => (
-          <option key={option.value} value={option.value}>
-            {option.label}
+    <form onSubmit={handleSubmit(onSubmit)}>
```
Comment (cgenardini) [inline src/components/form.Select/form.Select.tsx:44]: should be wrapped in the from component wrap in the form.stories not inside the components

File: src/components/form.Select/form.Select.tsx [hunk 3]
```diff
@@ -33,36 +34,46 @@ const FormSelect = ({
...
+            <option
+              className="font-text text-base font-normal leading-none "
+              key={index}
+              value={option.label}
+            >
+              {option.label}
+            </option>
+          ))}
+        </select>
+        {errors[name] && (
```
Comment (cgenardini) [inline src/components/form.Select/form.Select.tsx:71]: This should be more flexible to fit feedback messages - hint message or error message. See textInput or weekday components for reference.

**Review by cgenardini — COMMENTED** [review]
Great changes! Still needs some refactoring though. Look at the file structure in the main branch for the form components - textInput and weekday. The stories for all of the form components should live under the form.stories. They are both laid out very similar and will be good guidance for refactoring!  Also, can you change the name of the Pull request to match the component? I believe it has a name for a commit right now.

File: src/components/form.Select/form.Select.tsx (hunk 3 above)
Comment (kamalthapa1997) [reply src/components/form.Select/form.Select.tsx]: thankyou cgenardini, for your best suggestion. it helped me alot

**Pull request closed without merging**